### PR TITLE
Add code that has feature extraction of images

### PR DESCRIPTION
**Pull request by zaynpatel**
### What I did:

- Read images as a cv2 object
- Preprocessed them by turning them into grayscale, did edge detection so we have more features from the images to look at. Convert these to a 1D array so we can try to match feature vectors
- Use the knn algorithm to classify on images -- this was very bad. 0% accuracy. Paul suspects it's because raw feature matching w/vector similarity is extremely difficult. **Action item: Learn more about what the structure of the vector database is and how the input data should be structured. Raw images are probably not the best moving forward so we'll want to think of another way to get the data so it can be formatted for predict.**

### CODE REVIEW

**Comment by zaynpatel** [comment]
**Action item @MihirV17:** 
- What ML framework (PyTorch, TensorFlow) is best to work with images
- How should we train our images on a CNN and extract a vector with the features
- Rerun our knn logic with a CNN and see if we do better (goal is to have the CNN learn more features of each image)
